### PR TITLE
Default User Story + GuestBook/Passport + Like + Comment + Email + Service Worker Registration

### PR DESCRIPTION
**Pull request by kevinzhuang10**
#140 
#167 
#168 
#169
#170
#171

### CODE REVIEW

File: server/controllers/posts.js [hunk 1]
```diff
@@ -58,3 +65,22 @@ module.exports.getPostsByFollowings = (req, res) => {
         });
     });
 };
+
+module.exports.likePost = (req, res) => {
+  var {profile_id, post_id} = req.body;
+  models.Post.getPostById(post_id)
+    .then(post => {
+      post.profiles_likes().attach(profile_id);
```
Comment (lazdinst) [inline server/controllers/posts.js:73]: So Posts will have a likes property?

**Review by Sam0064 — APPROVED** [review]

File: server/app.js [hunk 2]
```diff
@@ -23,9 +24,33 @@ app.use(express.static(path.join(__dirname, '../public')));
 var upload = middleware.multer();
 app.use(upload.any());
 
+
 app.use('/', routes.auth);
 app.use('/api', routes.api);
 app.use('/api/profiles', routes.profiles);
 app.use('/upload', routes.upload);
 
+app.use('*', (req, res) => {
```
Comment (lazdinst) [inline server/app.js:33]: If we are injecting the preloaded state into the auth routes and this wild card route then we should break it out into a library helper function so we can update it all in one spot.

**Review by lazdinst — APPROVED** [review]

File: server/controllers/posts.js (hunk 1 above)
Comment (kevinzhuang10) [reply server/controllers/posts.js:73]: This is inserting into the likes join table. But we can certain have likes attached to posts

File: server/app.js (hunk 2 above)
Comment (kevinzhuang10) [reply server/app.js:33]: Yes totally agree

**Pull request merged**